feat(without): Add without function (#115)
toss/es-toolkit: docs/reference/array/without.md

# without

> 
> Creates an array excluding all given values using SameValueZero for equality comparisons.
> 
> This function takes an input array and returns a new array that excludes all values
> specified in the additional arguments. It uses SameValueZero for equality comparisons,
> meaning that it considers -0 and +0 as equal but treats NaN as unequal to itself.

Reviewer: ```suggestion

Creates an array that excludes all specified values.

It correctly excludes `NaN`, as it compares values using [SameValueZero](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-samevaluezero).
```

## Signature

```typescript

> function without<T>(array: ArrayLike<T>, ...values: T[]): T[];

Reviewer: ```suggestion
function without<T>(array: T[], ...values: T[]): T[];
```

```

### Parameters

> - `array` (`ArrayLike<T>`): The array to filter.
> - `values` (`...T[]`): The values to exclude.

Reviewer: ```suggestion
- `array` (`T[]`): The array to exclude values.
- `values` (`...T[]`): The values to exclude.
```

### Returns

(`T[]`) A new array without the specified values.

> ### Throws
> 
> Does not throw explicitly but returns an empty array if the input is not a valid array or has invalid array length.
> 

Reviewer: ```suggestion

```
